matching-service/src/routes/routes.ts

```diff
@@ -41,49 +42,54 @@ router.post("/start_matching", async (req : Request, res : Response) => {
-        //         userStore.removeUser(user.userToken);
-        //         return res.status(408).send({message: "No match found due to timeout"});
-        //     }
-        // });
-        return res.status(200).send({message: "Started Queueing"});
+        // Check if the user is already matching or matched
+        if (userStore.hasUser(user.id)) {
+            return res.status(409).send({message: "This user already matching."});
+        }
+        // Send the new user to the Kafka topic
+        userStore.addUser(user.id, user);
+        await sendQueueingMessage(user.id);
+        
+        return res.status(200).send({message: "Started Queueing User: " + user.email});
 
         // By default, no match is found
-        userStore.removeUser(user.userToken);
-        return res.status(204).send({message: "No match found"});
+        // userStore.removeUser(user.userToken);
+        // return res.status(204).send({message: "No match found"});
     }
     catch(error) {
         console.error("Error when trying to match:" + error);
-        userStore.removeUser(user.userToken);
+        userStore.removeUser(user.id);
         return res.status(500).send({message: "Failed to match user."});
     }
 });
 
+/**
+ * Check the matching state for the user
+ * 
+ * Request body should contain the user's id.
+ * 
+ * Response status:
+ * - 200: Match found
+ * - 204: Matching
+ * - 400: User ID not found
+ * - 400: User not found in the queue
+ * - 500: Error checking match status
+ */
 router.post("/check_state", async (req : Request, rsp : Response) => {
-    console.log("user cookies: " + req.cookies); // Log cookies to check
-    console.log("user cookies jwt: " + req.cookies.jwt); // Log cookies to check
     try {
-        const userToken = req.cookies.jwt;
-        if (!userToken) {
-            return rsp.status(400).send({message: "User token is not provided."});
+        const id = req.body.id;
+        if (!id) {
+            return rsp.status(400).send({message: "ID is not provided for checking status."});
         } else {
-            console.log('User Token:', userToken);
-            if(!userStore.hasUser(userToken)) {
-                return rsp.status(400).send({message: "This user does not exist in the matching service."});
+            if(!userStore.hasUser(id)) {
+                return rsp.status(400).send({message: "This user does not exist in the queue."});
             }
-            
-            console.log('User Exists:', userStore.hasUser(userToken));
-            const user = userStore.getUser(userToken);
+
+            const user = userStore.getUser(id);
             if(user!.matchedUser) {
+                userStore.removeUser(id);
+                console.log('Status: Match found for user:', user?.email);
                 return rsp.status(200).send({message: "match found"});
             } else {
                 return rsp.status(200).send({message: "matching"});
```

Comment:
i find this part weird, shouldnt they have different http statuses? Perhaps can try code 202 for the 'matching' part

![image](https://github.com/user-attachments/assets/0483b76f-cd5f-47b3-9475-02436f09556d)